packages/fxa-graphql-api/src/config.ts

```diff
@@ -6,43 +6,99 @@ import convict from 'convict';
 import fs from 'fs';
 import path from 'path';
 
-const conf = convict({
-  authHeader: {
-    default: 'authorization',
-    doc: 'Authentication header that should be logged for the user',
-    env: 'AUTH_HEADER',
-    format: String,
-  },
-  database: {
+export interface MySQLConfig {
+  database: string;
+  host: string;
+  password: string;
+  port: number;
+  user: string;
+}
+
+export interface RedisConfig {
+  host: string;
+  keyPrefix: string;
+  port: number;
+}
+
+function makeMySQLConfig(envPrefix: string, defaultDatabase: string) {
+  return {
     database: {
-      default: 'fxa',
+      default: defaultDatabase,
       doc: 'MySQL database',
-      env: 'DB_DATABASE',
+      env: envPrefix + '_MYSQL_DATABASE',
       format: String,
     },
     host: {
       default: 'localhost',
       doc: 'MySQL host',
-      env: 'DB_HOST',
+      env: envPrefix + '_MYSQL_HOST',
       format: String,
     },
     password: {
       default: '',
       doc: 'MySQL password',
-      env: 'DB_PASSWORD',
+      env: '_MYSQL_PASSWORD',
       format: String,
     },
     port: {
       default: 3306,
       doc: 'MySQL port',
-      env: 'DB_PORT',
+      env: '_MYSQL_PORT',
       format: Number,
     },
     user: {
       default: 'root',
       doc: 'MySQL username',
-      env: 'DB_USERNAME',
+      env: '_MYSQL_USERNAME',
+      format: String,
+    },
+  };
+}
+
+function makeRedisConfig(envPrefix: string, defaultPrefix: string) {
+  return {
+    host: {
+      default: 'localhost',
+      env: envPrefix + '_REDIS_HOST',
+      format: String,
+      doc: 'Url for redis host',
+    },
+    keyPrefix: {
+      default: defaultPrefix,
+      env: envPrefix + '_REDIS_KEY_PREFIX',
       format: String,
+      doc: 'Redis key prefix to use',
+    },
+    port: {
+      default: 6379,
+      env: envPrefix + '_REDIS_PORT',
+      format: 'port',
+      doc: 'Port for redis server',
+    },
+  };
+}
+
+const conf = convict({
+  authHeader: {
+    default: 'authorization',
+    doc: 'Authentication header that should be logged for the user',
+    env: 'AUTH_HEADER',
+    format: String,
+  },
+  authServer: {
+    url: {
+      doc: 'URL of fxa-auth-server',
+      env: 'AUTH_SERVER_URL',
+      default: 'http://localhost:9000/v1',
+    },
+  },
+  database: {
+    mysql: {
+      auth: makeMySQLConfig('AUTH', 'fxa'),
+      profile: makeMySQLConfig('PROFILE', 'fxa_profile'),
+    },
+    redis: {
+      profile: makeRedisConfig('PROFILE', 'fxa-profile'),
```

Comment:
Do we need access to profile-server's redis instance in the GQL server? As far as I can tell, it's only used for the hapi cache on the `profileCache.get` server method that provides data for the OIDC userinfo endpoint for RPs.

Most of that data comes from auth-server's `/account/profile` endpoint. Looks like avatar and display_name are profile-server local, but those could come out of the DB if we're accessing that directly?